src/main/java/com/exadel/sober/controllers/ReasonController.java

```diff
@@ -0,0 +1,42 @@
+package com.exadel.sober.controllers;
+
+import com.exadel.sober.models.Promise;
+import com.exadel.sober.models.Reason;
+import com.exadel.sober.repositories.PromiseRepository;
+import com.exadel.sober.repositories.ReasonRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@CrossOrigin
+@RequestMapping(path="/reason")
+public class ReasonController {
```

Comment:
I think you not need ReasonController. Reason cannot be updated or deleted separately from Promise. And getting reason by promiseId using additional call - not the best idea. Try to do all of that in one SQL query when you gets promises just join descriptions

Reply:
moved from promise controller and service nessesary functional